scylla-cql/src/value.rs

```diff
@@ -812,210 +847,271 @@ impl TryInto<time_03::Time> for CqlTime {
 #[derive(Clone, Debug, PartialEq)]
 #[non_exhaustive]
 pub enum CqlValue {
+    /// ASCII-only string.
     Ascii(String),
+    /// Boolean value.
     Boolean(bool),
+    /// Binary data of any length.
     Blob(Vec<u8>),
+    /// Counter value, represented as a 64-bit integer.
     Counter(Counter),
+    /// Variable-precision decimal.
     Decimal(CqlDecimal),
     /// Days since -5877641-06-23 i.e. 2^31 days before unix epoch
-    /// Can be converted to chrono::NaiveDate (-262145-1-1 to 262143-12-31) using as_date
+    /// Can be converted to chrono::NaiveDate (-262145-1-1 to 262143-12-31) using [TryInto].
     Date(CqlDate),
+    /// 64-bit IEEE-754 floating point number.
     Double(f64),
+    /// A duration with nanosecond precision.
     Duration(CqlDuration),
+    /// An empty value, which is distinct from null and is some DB legacy.
     Empty,
+    /// 32-bit IEEE-754 floating point number.
     Float(f32),
+    /// 32-bit signed integer.
     Int(i32),
+    /// 64-bit signed integer.
     BigInt(i64),
+    /// UTF-8 encoded string.
     Text(String),
-    /// Milliseconds since unix epoch
+    /// Milliseconds since unix epoch.
     Timestamp(CqlTimestamp),
+    /// IPv4 or IPv6 address.
     Inet(IpAddr),
+    /// A list of CQL values of the same types.
     List(Vec<CqlValue>),
+    /// A map of CQL values, whose all keys have the same type
+    /// and all values have the same type.
     Map(Vec<(CqlValue, CqlValue)>),
+    /// A set of CQL values of the same types.
     Set(Vec<CqlValue>),
+    /// A user-defined type (UDT) value.
+    /// UDT is composed of fields, each with a name
+    /// and an optional value of its own type.
     UserDefinedType {
+        /// Keyspace the type belongs to.
         keyspace: String,
+        /// Name of the user-defined type.
         name: String,
-        /// Order of `fields` vector must match the order of fields as defined in the UDT. The
-        /// driver does not check it by itself, so incorrect data will be written if the order is
-        /// wrong.
+        /// Fields of the user-defined type - (name, value) pairs.
         fields: Vec<(String, Option<CqlValue>)>,
     },
+    /// 16-bit signed integer.
     SmallInt(i16),
+    /// 8-bit signed integer.
     TinyInt(i8),
-    /// Nanoseconds since midnight
+    /// Nanoseconds since midnight.
     Time(CqlTime),
+    /// Version 1 UUID, generally used as a “conflict-free” timestamp.
     Timeuuid(CqlTimeuuid),
+    /// A tuple of CQL values of independent types each, where each element can be `None`
+    /// if the value is null. The length of the tuple is part of its CQL type.
     Tuple(Vec<Option<CqlValue>>),
+    /// Universally unique identifier (UUID) of any version.
     Uuid(Uuid),
+    /// Arbitrary-precision integer.
     Varint(CqlVarint),
+    /// A vector of CQL values of the same type.
+    /// The length of the vector is part of its CQL type.
     Vector(Vec<CqlValue>),
 }
```

Comment:
Unrelated to this PR: Vector type is a good use case for array deserialization that we discussed not long ago.

Reply:
100%.